packages/aws-cdk/lib/api/cxapp/stacks.ts

```diff
@@ -403,3 +412,39 @@ export interface Tag {
   readonly Key: string;
   readonly Value: string;
 }
+
+function _makeCdkMetadataAvailableCondition() {
+  const options = new Array<any>();
+  // Find out all regions known to support AWS::CDK::Metadata from the RegionInfo facts database
+  for (const region of regionInfo.Fact.regions) {
+    if (regionInfo.Fact.find(region, regionInfo.FactName.CDK_METADATA_RESOURCE_AVAILABLE) === 'YES') {
+      options.push({ 'Fn::Equals': [{ Ref: 'AWS::Region' }, region] });
+    }
+  }
+  return _fnOr(options);
+}
+
+/**
+ * This takes a bunch of operands and crafts an `Fn::Or` for those. Funny thing is `Fn::Or` requires
+ * at least 2 operands and at most 10 operands, so we have to... do this.
+ */
+function _fnOr(operands: any[]): any {
+  if (operands.length === 0) {
+    throw new Error('Cannot build `Fn::Or` with zero operands!');
+  }
+  if (operands.length === 1) {
+    return operands[0];
+  }
+  if (operands.length <= 10) {
+    return { 'Fn::Or': operands };
+  }
+  return _fnOr(_inGroupsOf(operands, 10).map(group => _fnOr(group)));
```

Comment:
This will fail again once there are more than 100 regions, because then there will be more than 10 groups and the top-level `Fn::Or` will fail.

For now this is okay, but this needs a tree-building algo to be generally correct.

Reply:
Err no. Because I make it groups of 10 groups of 10 then. I'll basically build up a 10-ary tree fractally if necessary.

Reply:
No - because the code will group fractally until all that's left is groups of 10 or less items (which are themselves groups of 10 or less items, or 10 or less "leaves").